### DISCUSSION TITLE
Using `Plot` as a complicated viewer/editor (e.g. 2D CAD)

### DISCUSSION THREAD

**Discussion by tgross35** [emilk/egui#3213] [Q&A] [answered]
I am quite a big fan of the looks and controls of egui's plots. I am thinking of using it for an editor application, but have two questions about how possible this is:

1. Are there any performance concerns for having thousands of items on the plot? I'm wondering if items that are too small to display or outside the viewport are still involved in rendering calculations sent to the GPU, or if egui handles pruning objects that aren't rendered.
2. How possible would it be to allow interacting with objects in the plot? Registering a click is the basic requirement (this could probably be done by manually calculating from mouse events) but something that can drag and drop would also be nice.

**Comment by tgross35** [comment]
It seems like https://github.com/emilk/egui/issues/1485 may be relevant to my first point

**Comment by YgorSouza** [comment, marked as answer]
Using Plot for that would probably be very hard, since that's not what it was designed for. It would be better if we had what is proposed in #1811. But even then, you would have to write the logic to decide what gets drawn on each frame according to the bounds of the viewbox. egui can't determine what is the bounding box of a widget without actually rendering it.
